compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/inlineclasses/MemoizedInlineClassReplacements.kt

```diff
@@ -50,7 +50,8 @@ class MemoizedInlineClassReplacements {
                 !it.hasMangledParameters ||
                         it.isSyntheticInlineClassMember ||
                         it.origin == IrDeclarationOrigin.LOCAL_FUNCTION_FOR_LAMBDA ||
-                        it.origin == IrDeclarationOrigin.DELEGATED_PROPERTY_ACCESSOR -> null
+                        it.origin == IrDeclarationOrigin.DELEGATED_PROPERTY_ACCESSOR ||
+                        it.origin == JvmLoweredDeclarationOrigin.POLYMORPHIC_SIGNATURE_INSTANTIATION -> null
```

Comment:
Note: in general, both backends generate garbage when a Java method takes an instance of an inline class as an argument.

```java
public class A {
    public int f(Z z) { return z.getX(); }
}
```
```kotlin
inline class Z(val x: Int)

fun main() {
    println(A().f(Z(1)))
}
```

-> call to `A."f-X5_cmG0":(LZ;)I` in JVM_IR and `Method A."f-IQRRRT4":(LZ;)I` in JVM. cc @sfs

Reply:
Please report an [issue](https://kotl.in/issue).

Reply:
[KT-35229](https://youtrack.jetbrains.com/issue/KT-35229)